product/opni/components/dialogs/EditClusterDialog.vue

```diff
@@ -0,0 +1,120 @@
+<script>
+import AsyncButton from '@/components/AsyncButton';
+import Card from '@/components/Card';
+import { exceptionToErrorsArray } from '@/utils/error';
+import LabeledInput from '@/components/form/LabeledInput';
+import KeyValue from '@/components/form/KeyValue';
+import { updateCluster } from '@/product/opni/utils/requests';
+
+export default {
+  components: {
+    Card,
+    KeyValue,
+    LabeledInput,
+    AsyncButton,
+  },
+  data() {
+    return {
+      id:     '',
+      name:   '',
+      labels: {},
+      errors: [],
+    };
+  },
+  methods: {
+    close() {
+      this.$modal.hide('edit-cluster-dialog');
+    },
+
+    open(cluster) {
+      this.$set(this, 'id', cluster.id);
+      this.$set(this, 'name', cluster.name);
+      this.$set(this, 'labels', cluster.labels.reduce((acc, label) => {
+        const [key, value] = label.split('=');
+
+        acc[key] = value;
+
+        return acc;
+      }, {}));
```

Comment:
Two notes:

1. Even though I like reduce I've found quite a few people have a hard time understanding it at first glance so I've opted to just use forEach instead for the sake of others.
2. I'm noticing a significant amount of string parsing. Is there a reason we can't get easy to consume json objects from the backend and pass that around instead? If not, can we move more of it into the models instead?